src/vs/editor/browser/widget/diffEditorWidget.ts

```diff
@@ -1577,16 +1607,17 @@ abstract class ViewZonesComputer {
 			}
 
 			// [PRODUCE] View zone(s) in original-side due to foreign view zone(s) in modified-side
-			while (modifiedForeignVZ.current && modifiedForeignVZ.current.afterLineNumber <= modifiedEndEquivalentLineNumber) {
+			while (modifiedForeignVZ.current && this.getActualLineNumber(modifiedForeignVZ.current.afterLineNumber, modifiedHiddenRanges) <= modifiedEndEquivalentLineNumber) {
 				let viewZoneLineNumber: number;
-				if (modifiedForeignVZ.current.afterLineNumber <= modifiedEquivalentLineNumber) {
-					viewZoneLineNumber = originalEquivalentLineNumber - modifiedEquivalentLineNumber + modifiedForeignVZ.current.afterLineNumber;
+				const actualAfterLineNumber = this.getActualLineNumber(modifiedForeignVZ.current.afterLineNumber, originalHiddenRanges);
+				if (actualAfterLineNumber <= originalEquivalentLineNumber) {
+					viewZoneLineNumber = modifiedEquivalentLineNumber - originalEquivalentLineNumber + actualAfterLineNumber;
```

Comment:
This should be a typo, the correct one is  
```suggestion
					viewZoneLineNumber = originalEquivalentLineNumber - modifiedEquivalentLineNumber + actualAfterLineNumber;
```